man/coap_io.txt.in

```diff
@@ -36,39 +49,92 @@ After setting up all the contexts, resources, endpoints sessions etc., the
+There are two methods of how to call *coap_io_process*().
 
-1. Have *coap_run_once*() called from within a while() loop.  Under idle
-conditions (no input traffic) *coap_run_once*() will then get called every
-_timeout_ms_, but more frequently if there is input traffic.
+1. Have *coap_io_process*() called from within a while() loop.  Under idle
+conditions (no input traffic) *coap_io_process*() will then get called every
+_timeout_ms_, but more frequently if there is input / retransmission traffic.
 
 2. Wait on the file descriptor returned by *coap_context_get_coap_fd*()
-using select() or an event returned by epoll_wait(). If 'read' is available on
-the file descriptor, call *coap_run_once*() with _timeout_ms_ set to
-COAP_RUN_NONBLOCK.  See EXAMPLES below.
-
-NOTE: This method is only available for environments that support epoll
-(mostly Linux) as libcoap will then be using epoll internally to process all
-the file descriptors of the different sessions.
-
-The *coap_io_process_with_fds*() function is the same as *coap_run_once*()
+using *select*() or an event returned by epoll_wait(). If 'read' is available on
+the file descriptor, call *coap_io_process*() with _timeout_ms_ set to
+COAP_RUN_NONBLOCK. +
+*NOTE*: This second method is only available for environments that support epoll
+(mostly Linux) with libcoap compiled to use *epoll* (the default) as libcoap
+will then be using *epoll* internally to process all the file descriptors of
+the different sessions.
+
+See EXAMPLES below.
+
+The *coap_io_process*() function is the primary function applications should
+use. There are internal functions that *coap_io_process*() calls which are
+available to use if absolutely necessary.  These internal functions and how to
+use them is different depending on whether libcoap has been compiled to use
+*epoll* (Linux systems only) or not.
+
+For *epoll* libcoap, *coap_io_process*() in simple terms calls
+*coap_io_prepare_epoll()*, does an *epoll_wait*() and then calls
+*coap_io_do_epoll*() if needed to make sure that all event based i/o has been
+completed.
+
+For *non-epoll* libcoap, *coap_io_process*() in simple terms calls
+*coap_io_prepare_io*() to set up sockets[], sets up all of the *select*()
+parameters based on the COAP_SOCKET_WANT* values in the sockets[], does a
+*select*(), updates the sockets[] with COAP_SOCKET_CAN_* as appropriate and
+then calls *coap_io_do_io*() to make sure that all current i/o has been
+completed.
+
+The *coap_io_prepare_epoll*() function for the specified _context_ will
+iterate through the endpoints and sessions to transmit any triggered observer
+responses as well as handling any timed out packet re-transmissions.  Returned,
+based on _now_, is the number of milli-secs needed to delay until the next
+time that *coap_io_prepare_epoll*() needs to get called.  After this call a
+*epoll_wait*() should done.
+
+The *void coap_io_do_epoll*() function for the specified _context_ will
```

Comment:
I think the return type of `coap_io_do_epoll()` is not necessary here (makes reading just more difficult).

Reply:
Corrected.